### PR TITLE
Simplify CI builds

### PR DESCRIPTION
**Pull request by lgeiger**
This PR will reduce the build matrix, simplify the build scripts and therefor speed up the builds.

Originally I had in mind to setup a 32-bit Linux build. But we would need to do the builds inside a docker container which I'm not familiar with.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,33 +9,31 @@ addons:
...
-
+matrix:
+  include:
+    - node_js: "6"
+      env: DEPLOY="true"
+    - node_js: "6"
+      env: ELECTRON="1.1.3"
+    - node_js: "6"
+      env: ELECTRON="1.4.4" REBUILD="true"
+    - os: osx
```
Comment (rgbkrk) [inline .travis.yml]: Oh cool so we'll still get prebuild for macOS / OS X then?

Comment (lgeiger) [inline .travis.yml]: Yes: https://travis-ci.org/zeromq/zeromq.js/builds/171775762

**Comment by rgbkrk** [comment]
I can certainly handle creating a Docker image, I haven't tried making a 32 bit build before though.

**Comment by willingc** [comment]
@lgeiger Quick question for my own education re: Travis: Any particular reason to go for the 32 bit build?

**Comment by rgbkrk** [comment]
We currently do prebuilt binaries for Windows 32 and 64 bit, Linux 64 bit, and OS X/macOS 64 bit. @n-riesco requested 32 bit Linux, and all we could do at the time was making sure building from source worked well.

I was indeed surprised that people were still getting 32 bit machines and that you can still purchase windows machines that are 32 bit.

**Comment by lgeiger** [comment]
@willingc Currently we ship prebuilt node binaries for OS X 64-bit, Windows 64-bit, Windows 32-bit and Linux 64-bit.

If we setup a 32-bit build we could also ship Linux 32-bit binaries. The problem is Travis and Circle CI only have 64-bit VMs. This means we'd need to build inside a docker container on Travis.

Do you know of a build service supporting 32-bit linux builds?

**Comment by lgeiger** [comment]
@rgbkrk Ahh you where faster 😂

**Pull request merged**